### PR TITLE
Appium enter, clear, clear then enter text

### PR DESCRIPTION
**Pull request by ncalaway**
Can now enter text into an input field.

### CODE REVIEW

**Comment by ncalaway** [comment]
Can now enter text, clear text, and clear then enter text.

File: src/appium/appiumelement.ts [hunk 1]
```diff
@@ -55,6 +55,38 @@ export class AppiumElement extends DOMElement implements iValue {
...
+      {
+        method: "post",
+      }
+    );
+  }
+
+  public async clearThenType(input: string): Promise<void> {
+    await this.clear();
+    await this.type(input);
+  }
```
Comment (johnsickels) [inline src/appium/appiumelement.ts:88]: Is an error handled the same way if we implicitly return a promise vs:

```
public async clearThenType(input: string): Promise<void> {
  await this.clear();
  return this.type(input);
}
```
The return type will still be `Promise<void>`, I was just wondering if there are any differences to look out for?

**Review by johnsickels — APPROVED** [review]

File: src/appium/appiumelement.ts (hunk 1 above)
Comment (ncalaway) [reply src/appium/appiumelement.ts:88]: Actually, neither way handles errors correctly. If I return a Promise<void> at all, the error is never returned if an element cannot be typed into.

Comment (ncalaway) [reply src/appium/appiumelement.ts:88]: Appium throws errors if you try to type into an element that doesn't support an interaction, and I'm not sure how to capture those. I'm not allowed to return a `Promise<iValue>` from the `type()` method.

Comment (ncalaway) [reply src/appium/appiumelement.ts:88]: I was considering just leaving it out since you don't do assertions on typing or clearing text. What do you think?

Comment (johnsickels) [reply src/appium/appiumelement.ts:88]: We don't do error handling with Puppeteer, but I wonder if there is an opportunity here to improve that. If the element doesn't exist or can't be typed into, should we just let the Node process fail, or could we gracefully stop the suite and print a pretty error to the console?

Comment (ncalaway) [reply src/appium/appiumelement.ts:88]: I can do a `throw` if the Appium JSON response contains an `error` property. How would that work for Puppeteer?

Comment (johnsickels) [reply src/appium/appiumelement.ts:88]: If we `throw` isn't that still an unhandled promise rejection? I don't have any better suggestions, just spit ballin here

Comment (johnsickels) [reply src/appium/appiumelement.ts:88]: Cool, as long as we have a failing scenario in mind. Thanks for entertaining my questions 😃

**Pull request merged**